[Feat] Post 도메인 공통 개발

ohyeryung opened the pull request:
## 📌 작업 내용 (필수)
- Post 관련 도메인, 서비스, 레포지토리 생성하였습니다.

<br/>

## 🌱 반영 브랜치

- feat/#ALL-37-domain-post -> dev
- close #16

<br/>

ohyeryung committed:
[Feat] Post 관련 도메인 개발

# [Header]
# 형식 - [Type] title (#이슈번호)
#
# [Type] 입력 목록
#  1. Feat: 새로운 기능 추가
#  2. Fix: 오류 해결
#  3. Rename: 파일 혹은 폴더명을 수정만 한 경우
#  4. Remove: 파일을 삭제만 한 경우
#  5. Style: 기능 수정 없이 코드 스타일만 변경한 경우 (코드 포매팅, 세미콜론 누락 등)
#  6. Design: 사용자 UI 디자인 변경 (CSS 등)
#  7. Refactor: 코드 베이스의 특정 부분을 재정렬 (Refactoring)
#  8. Test: 테스트와 관련된 모든 것
#  9. Docs: 문서화에 관한 모든 것
#  10. Chore: 빌드 업무 수정, 패키지 매니저 수정 (gitignore 수정 등)
#  11. Perf: 성능 개선
#
# [Title] 규칙
#  1. 50자 이내 짧은 요약 (영어의 경우 첫글자는 대문자로)
#  2. 명령형으로 작성 (과거형X)
#  3. 제목 끝에 .(마침표) 금지
#
# [Body] 작성 규칙
#  1. 제목과 본문 사이 빈 칸 한줄 필수 (제목과 본문 분리)
#  2. 생략 가능, 필요 시 자세한 설명 작성 (72자 내외로 마무리)
#  3. 적용한 변경사항과 그것을 만든 이유에 대해 설명
#     - 어떻게보다는 "무엇을", "왜"

ohyeryung committed:
[Feat] Post 관련 컨트롤러, 서비스, 레포지토리 추가

# [Header]
# 형식 - [Type] title (#이슈번호)
#
# [Type] 입력 목록
#  1. Feat: 새로운 기능 추가
#  2. Fix: 오류 해결
#  3. Rename: 파일 혹은 폴더명을 수정만 한 경우
#  4. Remove: 파일을 삭제만 한 경우
#  5. Style: 기능 수정 없이 코드 스타일만 변경한 경우 (코드 포매팅, 세미콜론 누락 등)
#  6. Design: 사용자 UI 디자인 변경 (CSS 등)
#  7. Refactor: 코드 베이스의 특정 부분을 재정렬 (Refactoring)
#  8. Test: 테스트와 관련된 모든 것
#  9. Docs: 문서화에 관한 모든 것
#  10. Chore: 빌드 업무 수정, 패키지 매니저 수정 (gitignore 수정 등)
#  11. Perf: 성능 개선
#
# [Title] 규칙
#  1. 50자 이내 짧은 요약 (영어의 경우 첫글자는 대문자로)
#  2. 명령형으로 작성 (과거형X)
#  3. 제목 끝에 .(마침표) 금지
#
# [Body] 작성 규칙
#  1. 제목과 본문 사이 빈 칸 한줄 필수 (제목과 본문 분리)
#  2. 생략 가능, 필요 시 자세한 설명 작성 (72자 내외로 마무리)
#  3. 적용한 변경사항과 그것을 만든 이유에 대해 설명
#     - 어떻게보다는 "무엇을", "왜"

bsjin1122 commented on src/main/java/com/allclear/socialhub/post/domain/Post.java:
```diff
@@ -1,4 +1,44 @@
+    private PostType postType;
+
+    @Column(nullable = false)
+    private int viewCnt;
```
nullable false면 default=0필요하지 않을까요?! 물론 기본형타입이라 int는 0이 기본으로 들어가긴 하는데 헷갈려서.. 남겨봅니다

ryuneng replied:
필드에 default 값 적용하기 위해 예령님과 여러 방면으로 의논해보았는데요.
JPA는 데이터베이스에 기본 값을 설정하는 기능을 직접적으로 제공하지 않아서 데이터베이스 레벨에서 설정해야 하더라구요. 어떤 방법이 적합할지 모르겠어서 좀더 고민해보고 수정하겠습니다!

아래 목록처럼 여러 방법을 고민해보았으나, 적합하지 않은 것 같아 제외했습니다.
1. @Column(nullable = false) 설정 시 Java 단에서 자동으로 0으로 초기화
 - 데이터베이스의 제약 조건을 직접적으로 제어하지 않기 때문에 데이터 무결성을 위해 데이터베이스 자체에서 기본값을 설정하는 것이 좋음
2. private int viewCnt = 0; 으로 필드 초기값 설정
 - 1번과 동일
3. @ColumnDefault / @Column의 columnDefinition 속성
 - ddl-auto 설정값에 따라 오류 발생 가능성

ohyeryung replied:
회의 때 논의 후 정리 완료하였습니다~

bsjin1122 commented on src/main/java/com/allclear/socialhub/post/common/hashtag/domain/HashTag.java:
```diff
@@ -0,0 +1,16 @@
+    private Long id;
+
+    @Column(nullable = false)
+    private String content;
```
해시태그 글자수(내용) 제한은 없으신거죠!? 확인차 다시 여쭤봅니다

ohyeryung replied:
한 해시태그 당 길이 제한 추가하면 좋을 것 같아서요, 이것도 수정하는 대로 커밋 올리겠습니다~

bsjin1122 commented on src/main/java/com/allclear/socialhub/post/common/view/repository/ViewRepository.java:
```diff
@@ -0,0 +1,7 @@
+import com.allclear.socialhub.post.common.view.domain.PostView;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface ViewRepository extends JpaRepository<PostView, Long> {
```
엔터티명과 맞춰서 PostViewRepository는 어떨까 싶어요!

ohyeryung replied:
네 좋습니다! 반영해서 다시 커밋하겠습니다~

bsjin1122 reviewed (commented):
코멘트 드립니다

bsjin1122 commented on src/main/java/com/allclear/socialhub/post/common/share/repository/ShareRepository.java:
```diff
@@ -0,0 +1,7 @@
+import com.allclear.socialhub.post.common.share.domain.PostShare;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface ShareRepository extends JpaRepository<PostShare, Long> {
```
여기도 PostShareRepository 의견드립니다~

ohyeryung replied:
네 확인했습니다!

bsjin1122 reviewed (commented):
2번째 코멘트 일괄? 드립니다

bsjin1122 commented:
수정 사항이 많이 발생할 수 있으니, 섣부른?(괜찮다고 판단하셔서 누르신거겠지만!) approve 보다 타 기능 맡으시는 분들이 approve 를 눌러주시는 건 어떨까 생각이 들었습니다. (지금은 서로 다른 파트로 2명씩 짝이 되었으니 나머지 4분들 중 2명이 승인? 코드를 본다는 차원에서의 의견) 
- 이것도 회의에 안건으로 .. 생각이 드신다면..! 회의록 안건에 넣어주셔도 상관없습니다.
- 그나저나 저는 왜 승인 버튼이 없는지 아직 잘 몰라서, 다시 찾아보고도 안되면 화면공유 요청드리겠습니다.

ryuneng commented:
> 수정 사항이 많이 발생할 수 있으니, 섣부른?(괜찮다고 판단하셔서 누르신거겠지만!) approve 보다 타 기능 맡으시는 분들이 approve 를 눌러주시는 건 어떨까 생각이 들었습니다. (지금은 서로 다른 파트로 2명씩 짝이 되었으니 나머지 4분들 중 2명이 승인? 코드를 본다는 차원에서의 의견)
> 
> * 이것도 회의에 안건으로 .. 생각이 드신다면..! 회의록 안건에 넣어주셔도 상관없습니다.
> * 그나저나 저는 왜 승인 버튼이 없는지 아직 잘 몰라서, 다시 찾아보고도 안되면 화면공유 요청드리겠습니다.

ohyeryung committed:
[Rename] Post 관련 레포지토리명 수정

# [Header]
# 형식 - [Type] title (#이슈번호)
#
# [Type] 입력 목록
#  1. Feat: 새로운 기능 추가
#  2. Fix: 오류 해결
#  3. Rename: 파일 혹은 폴더명을 수정만 한 경우
#  4. Remove: 파일을 삭제만 한 경우
#  5. Style: 기능 수정 없이 코드 스타일만 변경한 경우 (코드 포매팅, 세미콜론 누락 등)
#  6. Design: 사용자 UI 디자인 변경 (CSS 등)
#  7. Refactor: 코드 베이스의 특정 부분을 재정렬 (Refactoring)
#  8. Test: 테스트와 관련된 모든 것
#  9. Docs: 문서화에 관한 모든 것
#  10. Chore: 빌드 업무 수정, 패키지 매니저 수정 (gitignore 수정 등)
#  11. Perf: 성능 개선
#
# [Title] 규칙
#  1. 50자 이내 짧은 요약 (영어의 경우 첫글자는 대문자로)
#  2. 명령형으로 작성 (과거형X)
#  3. 제목 끝에 .(마침표) 금지
#
# [Body] 작성 규칙
#  1. 제목과 본문 사이 빈 칸 한줄 필수 (제목과 본문 분리)
#  2. 생략 가능, 필요 시 자세한 설명 작성 (72자 내외로 마무리)
#  3. 적용한 변경사항과 그것을 만든 이유에 대해 설명
#     - 어떻게보다는 "무엇을", "왜"

ohyeryung committed:
[Fix] 해시태그 길이 제한

ohyeryung commented:
좋은 의견입니다! 저는 근데 같은 파트 맡으신 분이 승인해주시는 것도 필요하다고 생각이 들어서요!
아무래도 같은 부분을 개발하고 구조도 더 잘 이해하고 있을 거라고 생각이 들어서 더더욱 코드 리뷰를 해야한다고 생각했습니다.

ryuneng commented:
좋은 의견 감사합니다. 제가 생각한 방향은 위의 예령님 의견과 동일하여 회의 때 의논해보면 좋을 것 같습니다!
추가적으로 승인 버튼은 Files Changed 탭 > 초록색 Review Changes 버튼을 눌렀을 때 Approve 선택값이 안보이시는 걸까요?

![스크린샷 2024-08-23 024630](https://github.com/user-attachments/assets/15318ee3-637d-4f17-ab43-c0e058d71404)
